themes-default/slim/views/editShow.mako

```diff
@@ -43,121 +42,133 @@ const startVue = () => {
                     },
                     language: 'en'
                 },
-                defaultEpisodeStatusOptions: [
-                    {text: 'Wanted', value: 'Wanted'},
-                    {text: 'Skipped', value: 'Skipped'},
-                    {text: 'Ignored', value: 'Ignored'}
-                ],
-                seriesLoaded: false,
+                showLoaded: false,
                 saving: false
             }
         },
-        async mounted() {
-            const seriesSlug = $('#series-slug').attr('value');
-            const url = 'series/' + seriesSlug;
-            try {
-                const response = await api.get('series/' + this.seriesSlug);
-                this.series = Object.assign({}, this.series, response.data);
-                this.series.language = response.data.language;
-            } catch (error) {
-                console.debug('Could not get series info for: '+ seriesSlug);
+        computed: Object.assign(store.mapState(['shows']), {
+            params() {
+                return location.search.slice(1).split('&').reduce((obj, pair) => {
+                    const [ key, value ] = pair.split('=');
+                    obj[key] = value;
+                    return obj;
+                }, {});
+            },
+            id() {
+                return this.params.seriesid;
+            },
+            indexer() {
+                return this.params.indexername;
+            },
+            // @TODO: Enable this once we remove this.series
+            // show() {
+            //     const { $store } = this;
+            //     return this.shows.length === 0 ? $store.defaults.show : this.shows.find(show => show.indexer === this.indexer && Number(show.id[show.indexer]) === Number(this.id));
```

Comment:
1. This wouldn't work as is (`$store` is supposed to be `$store.state`).
2. I'd prefer if you split it into multiple lines.
3. I have a suggestion.
```js
            show() {
                const { $store, id, indexer } = this;
                const { state } = $store;
                if (this.shows.length === 0 || !indexer || !id) return state.defaults.show;
                const slug = indexer + id.toString();  // or String(id)
                return this.shows.find(show => show.id.slug === slug);
            },
```

And lastly: Wouldn't we still need some form of a `show` object anyway?
What object will be updated when the users changes the values?
I'm guessing here that it would be the new `show` object in the store state.
But if that's the case you won't be able to have 2 properties with the same name... So, how?